packages/keycloak-user-management/src/components/UserGroupDetailView/tests/index.test.tsx

```diff
@@ -0,0 +1,201 @@
+import { mount } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import React from 'react';
+import { Router } from 'react-router';
+import { store } from '@opensrp/store';
+import { authenticateUser } from '@onaio/session-reducer';
+import * as notifications from '@opensrp/notifications';
+import fetch from 'jest-fetch-mock';
+import { ViewDetails } from '..';
+import * as fixtures from './fixtures';
+import { createBrowserHistory } from 'history';
+import { URL_USER_GROUPS } from '../../../constants';
+import { act } from 'react-dom/test-utils';
+import flushPromises from 'flush-promises';
+
+const history = createBrowserHistory();
+
+jest.mock('@opensrp/notifications', () => ({
+  __esModule: true,
+  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
+}));
+
+describe('View User Group Details', () => {
+  beforeAll(() => {
+    store.dispatch(
+      authenticateUser(
+        true,
+        {
+          email: 'bob@example.com',
+          name: 'Bobbie',
+          username: 'RobertBaratheon',
+        },
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        { api_token: 'hunter2', oAuth2Data: { access_token: 'sometoken', state: 'abcde' } }
+      )
+    );
+  });
+
+  afterEach(() => {
+    fetch.mockClear();
+  });
+
+  it('works correctly', async () => {
+    fetch.once(JSON.stringify(fixtures.members)).once(JSON.stringify(fixtures.userGroup1));
+    const props = {
+      groupId: fixtures.userGroup1.id,
+      keycloakBaseURL:
+        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
+    };
+
+    const wrapper = mount(
+      <Router history={history}>
+        <ViewDetails {...props} />
+      </Router>
+    );
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+    });
+
+    expect(wrapper.text()).toMatchSnapshot('nominal display');
```

Comment:
I think we need a snapshot here (or assertion statements) to capture the UI elements say
If another developer edits something else such as the `Space` from  `direction="vertical"` to `direction="horizontal" the tests will still pass